src/meta.ts

```diff
@@ -42,6 +44,11 @@ export class Meta {
     this.flavor = fcl.Transform(inputs.flavor);
     this.date = new Date();
     this.version = this.getVersion();
+    this.commit_date = this.getCommitDate(this.context.sha);
+  }
+
+  private async getCommitDate(rev: string): Promise<Date> {
+    return await Git.commitDate(rev);
```

Comment:
Should only be invoked for `git` context, not workflow one, see https://github.com/docker/metadata-action/pull/308#discussion_r1262479919

Reply:
Yes I understand. It's a draft, I'm still changing it

Reply:
should I add a new propert to `Context`?

Reply:
for example, `Context = GithubContext & {commitDate:Date}`

Reply:
> should I add a new propery to `Context`?

`Context` is a GitHub object and does not expose ref date metadata: https://github.com/docker/actions-toolkit/blob/8672cc70f4a00ce6c373b71b289ae74483b3854a/src/git.ts#L28

But we could extend it in https://github.com/docker/metadata-action/blob/master/src/context.ts to add this property if this is what you were thinking of.

Reply:
for example, for a PR, commit is generated by github and `commits` doesn't exists in event.